pkg/utils/imagevector/imagevector.go

```diff
@@ -79,6 +79,25 @@ func (v ImageVector) FindImage(name, k8sVersionRuntime, k8sVersionTarget string)
 	return nil, fmt.Errorf("could not find image %q for Kubernetes runtime version %q in the image vector", name, k8sVersionRuntime)
 }
 
+// FindImages returns an image map with the given <names> from the sources in the image vector.
+// The <k8sVersion> specifies the kubernetes version the image will be running on.
+// The <targetK8sVersion> specifies the kubernetes version the image shall target.
+// If multiple entries were found, the provided <k8sVersion> is compared with the constraints
+// stated in the image definition.
+// In case multiple images match the search, the first which was found is returned.
+// In case no image was found, an error is returned.
+func (v ImageVector) FindImages(names []string, k8sVersionRuntime, k8sVersionTarget string) (map[string]interface{}, error) {
+	images := make(map[string]interface{}, len(names))
```

Comment:
I wouldn't pre-allocate here since this method might fail / there might be duplicate names. Only pre-allocate with a given size if you definitely know the outcome / definitely know a minimum value.